### DISCUSSION TITLE
How integrate web worker correctly ? [Typescript]

### DISCUSSION THREAD

**Discussion by binaryangel-noa** [jaredpalmer/razzle#1774] [Q&A]
Hello. 
I am looking for strategies / examples / guides about how to integrate Web Worker into a razzle project written in TypeScript

Thanks ahead! :-) 🦊

**Comment by fivethreeo** [comment]
https://github.com/jaredpalmer/razzle/issues/1544

**Comment by fivethreeo** [comment]
https://v4.webpack.js.org/loaders/worker-loader/

**Reply by binaryangel-noa** [reply]
Based on this i almost got where i wanted... but one the last mile i have an issue that might be CORS related ?
I receive the error [Uncaught DOMException: Failed to construct 'Worker': Script at 'http://localhost:3001/static/js/firebase.worker.worker.js' cannot be accessed from origin 'http://localhost:3000'."]

When i try to access it directly, the file is there. And npm run build + npm run start:prod works correctly too.

Did you happen to come across something like that ?

**Comment by fivethreeo** [comment]
Yeah, cors i think

**Reply by binaryangel-noa** [reply]
Any idea what to look for to resolve it ? Missing any clue currently... hmm... might be enough to add a cors rule to the express server ?

**Reply by fivethreeo** [reply]
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/worker-src

**Reply by fivethreeo** [reply]
yes, but needs to be a specific csp, not wildcard

**Comment by fivethreeo** [comment]
I would be interested in seeing how you set this up, possible to share how?

**Reply by binaryangel-noa** [reply]
I will try to... currently its abit of mess... was looking into a "good" way to integrate firebase messaging push notifications :-D
